app/assets/javascripts/modules/gtm-track-copy.js

```diff
@@ -0,0 +1,35 @@
+function GTMCopyListener ($module, dataLayer) {
+  this.$module = $module
+  this.dataLayer = dataLayer || window.dataLayer
+}
+
+GTMCopyListener.prototype.handleCopyEvent = function (component) {
+  this.dataLayer.push({
+    'event': 'text-copied',
+    'component': component
+  })
+}
+
+GTMCopyListener.prototype.handlePasteEvent = function (component) {
+  this.dataLayer.push({
+    'event': 'text-pasted',
+    'component': component
+  })
+}
+
+GTMCopyListener.prototype.init = function () {
+  var $module = this.$module
+  var moduleName = this.$module.dataset.module
+  $module.addEventListener('copy', function (e) {
+    this.handleCopyEvent(moduleName)
+  }.bind(this), false)
+
+  $module.addEventListener('paste', function (e) {
+    this.handlePasteEvent(moduleName)
+  }.bind(this), false)
+}
+
+var $trackedComponents = document.querySelectorAll('[data-module="markdown-editor"], [data-module="copy-to-clipboard"]')
```

Comment:
Rather than adding this to an explicit selector could we add this as a data attribute that could be selected on - in a similar vein to: https://github.com/alphagov/content-publisher/blob/b161ecb623c642e6fc3199e619de320b9adca5c4/app/assets/javascripts/modules/warn-before-unload.js#L26